openedx/core/djangoapps/credit/tasks.py

```diff
@@ -208,17 +216,23 @@ def _get_proctoring_requirements(course_key):
     # process
     from edx_proctoring.api import get_all_exams_for_course
 
-    requirements = [
-        {
-            'namespace': 'proctored_exam',
-            'name': exam['content_id'],
-            'display_name': exam['exam_name'],
-            'criteria': {},
-        }
-        for exam in get_all_exams_for_course(unicode(course_key))
-        # practice exams do not count towards eligibility
-        if exam['is_proctored'] and exam['is_active'] and not exam['is_practice_exam']
-    ]
+    requirements = []
+    for exam in get_all_exams_for_course(unicode(course_key)):
+        if exam['is_proctored'] and exam['is_active'] and not exam['is_practice_exam']:
+            try:
+                usage_key = UsageKey.from_string(exam['content_id'])
+                proctored_block = modulestore().get_item(usage_key)
```

Comment:
We already retrieved the proctored exam blocks earlier when creating the proctored exam record (in `contentstore.proctoring.register_proctored_exams`).  Is there any way we could include the `start_date` when registering the proctored exam, so that:

1) We avoid making additional Mongo queries in the credit task, and
2) We avoid making assumptions about how proctored exams are represented in the courseware.